thousandeyes/resource_alert_rule.go

```diff
@@ -45,27 +45,11 @@ func resourceAlertRuleUpdate(d *schema.ResourceData, m interface{}) error {
 
 	log.Printf("[INFO] Updating ThousandEyes Test %s", d.Id())
 	id, _ := strconv.ParseInt(d.Id(), 10, 64)
-	update := ResourceUpdate(d, &thousandeyes.AlertRule{}).(*thousandeyes.AlertRule)
 	// While most ThousandEyes updates only require updated fields and specifically
-	// disallow some fields on update, Alert Rules actually require a few fields
-	// to be retained on update.
-	// Terraform schema validation should guarantee their existence.
-	update.AlertType = thousandeyes.String(d.Get("alert_type").(string))
-	update.Expression = thousandeyes.String(d.Get("expression").(string))
-
-	// MinimumSources and MinimumSourcesPct are mutually exclusive
-	minimumSources := d.Get("minimum_sources").(int)
-	if minimumSources > 0 {
-		update.MinimumSources = thousandeyes.Int(minimumSources)
-	} else {
-		update.MinimumSourcesPct = thousandeyes.Int(d.Get("minimum_sources_pct").(int))
-	}
```

Comment:
Since we're sending the entire list of fields, I would argue that we don't need this logic here. If someone sets both fields by chance then the API will return the error which will be passed to the provider. Let me know if you disagree.

Reply:
i agree that it's better to have the validation logic in one place.